pkg/buses/message_dispatcher.go

```diff
@@ -86,30 +92,37 @@ func (d *MessageDispatcher) executeRequest(url *url.URL, message *Message) (*Mes
 	d.logger.Infof("Dispatching message to %s", url.String())
 	req, err := http.NewRequest(http.MethodPost, url.String(), bytes.NewReader(message.Payload))
 	if err != nil {
-		return nil, fmt.Errorf("Unable to create request %v", err)
+		return nil, fmt.Errorf("unable to create request %v", err)
 	}
 	req.Header = d.toHTTPHeaders(message.Headers)
 	res, err := d.httpClient.Do(req)
 	if err != nil {
 		return nil, err
 	}
-	if res != nil {
-		if res.StatusCode < 200 || res.StatusCode >= 300 {
-			// reject non-successful (2xx) responses
-			return nil, fmt.Errorf("unexpected HTTP response, expected 2xx, got %d", res.StatusCode)
-		}
-		headers := d.fromHTTPHeaders(res.Header)
-		// TODO: add configurable whitelisting of propagated headers/prefixes (configmap?)
-		if correlationID, ok := message.Headers[correlationIDHeaderName]; ok {
-			headers[correlationIDHeaderName] = correlationID
-		}
-		payload, err := ioutil.ReadAll(res.Body)
-		if err != nil {
-			return nil, fmt.Errorf("Unable to read response %v", err)
-		}
-		return &Message{headers, payload}, nil
+	if res == nil {
+		// I don't think this is actually rechable with http.Client.Do(), but just to be sure we
+		// check anyway.
+		return nil, nil
```

Comment:
`nil, nil` is a surprising result -- if we think this is unreachable, would returning a `nil, error.New("non-error nil result from http.Client.Do()")` make more sense?

Reply:
I agree, done.